project-installcheck.py

```diff
@@ -97,22 +125,193 @@ def project_repository(self, project):
+
+            parsed = parsed_installcheck(pfile, arch, target_packages, [])
+            for package in parsed:
+                parsed[package]['output'] = "\n".join(parsed[package]['output'])
+
+            # let's risk a N*N algorithm in the hope that we have a limited N
+            for package1 in parsed:
+                output = parsed[package1]['output']
+                for package2 in parsed:
+                    if package1 == package2:
+                        continue
+                    output = output.replace(parsed[package2]['output'], 'FOLLOWUP(' + package2 + ')')
+                parsed[package1]['output'] = output
+
+            for package in parsed:
+                parsed[package]['output'] = self._split_and_filter(parsed[package]['output'])
+
+        url = makeurl(self.apiurl, ['build', project, '_result'], {
+                      'repository': repository, 'arch': arch, 'code': 'succeeded'})
+        root = ET.parse(http_GET(url)).getroot()
+        succeeding = list(map(lambda x: x.get('package'), root.findall('.//status')))
+
+        per_source = dict()
+
+        for package, entry in parsed.items():
+            source = "{}/{}/{}/{}".format(project, repository, arch, entry['source'])
+            per_source.setdefault(source, {'output': [], 'builds': entry['source'] in succeeding})
+            per_source[source]['output'].extend(entry['output'])
+
+        rebuilds = set()
+
+        for source in sorted(per_source):
+            if not len(per_source[source]['output']):
+                continue
+            self.logger.debug("{} builds: {}".format(source, per_source[source]['builds']))
+            self.logger.debug("  " + "\n  ".join(per_source[source]['output']))
+            if not per_source[source]['builds']:  # nothing we can do
+                continue
+            old_output = oldstate['check'].get(source, {}).get('problem', [])
+            if old_output == per_source[source]['output']:
+                self.logger.debug("unchanged problem")
+                continue
+            self.logger.info("rebuild %s", source)
+            rebuilds.add(os.path.basename(source))
+            for line in difflib.unified_diff(old_output, per_source[source]['output'], 'before', 'now'):
+                self.logger.debug(line.strip())
+            oldstate['check'][source] = {'problem': per_source[source]['output'],
+                                         'rebuild':  str(datetime.datetime.now())}
+
+        for source in list(oldstate['check']):
+            if not source.startswith('{}/{}/{}/'.format(project, repository, arch)):
+                continue
+            if not os.path.basename(source) in succeeding:
+                continue
+            if source not in per_source:
+                self.logger.info("No known problem, erasing %s", source)
+                del oldstate['check'][source]
+
+        packages = ['branding-openSUSE', 'PackageKit-branding-openSUSE', 'xfce4-branding-openSUSE',
```

Comment:
Should we make this into a configuration inside OBS/project as well? From the looks of it, I'd miss at least installation-images:MicroOS right now

Reply:
Yes, we should.

Reply:
Feel free to put it into osrt:config already for later pickup. But remember to do it for the ports as well